sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs

```diff
@@ -212,7 +223,7 @@ public BlobServiceClient(Uri serviceUri, TokenCredential credential, BlobClientO
         /// every request.
         /// </param>
         internal BlobServiceClient(Uri serviceUri, HttpPipelinePolicy authentication, BlobClientOptions options)
-            : this(serviceUri, authentication, options.Build(authentication), new ClientDiagnostics(options))
+            : this(serviceUri, authentication, options.Build(authentication), new ClientDiagnostics(options), options?.CustomerProvidedKey)
```

Comment:
Can options be null here? If so, there would already be a bug in the options.Build call.

Reply:
No - everyone who calls this does `options ?? new BlobClientOptions()`.

Reply:
So can you update to be options.CustomerProvidedKey?

Reply:
I could, but this is still safer so I'd prefer to leave it.